### ISSUE TITLE
Support custom created token / support two-factor authentication

### ISSUE THREAD

**Issue by jzaefferer** [arboleya/ghfm#3]
Since activating two-factor authentication on GitHub, I can't use this tool anymore. When trying to render a markdown file, I'm asked to enter username and password, which always gets rejected with `Incorrect credentials, try again`. I guess it just doesn't know about two-factor authentication and so outputs a misleading error message?

It would be great if I could just create a token manually at https://github.com/settings/applications and use that.

**Comment by arboleya** [comment]
You're right, it doesn't know about 2FA, there's more than one option for working around this.
https://developer.github.com/changes/2013-09-03-two-factor-authentication/

In this meanwhile, you can work around it by putting your token inside the file:
- `ghfm/lib/user.oauth.json`

Try this one, put your token and run it:

``` shell
echo '{"token": "YOUR_TOKEN_HERE"}' > $(dirname $(which ghfm))/../lib/node_modules/ghfm/lib/user.oauth.json
```

Btw, I wasn't even aware that there was somebody using this lib. Glad to see there is. :)

Let me know if it worked.

**Comment by jzaefferer** [comment]
Yes, that worked! When creating the token, there's a bunch of checkboxes to select what this token can be used for. I left it at the defaults, that worked fine.

**Comment by arboleya** [comment]
Thanks for the feedback, I'm going to look for an efficient way to get through this.

**Comment by jzaefferer** [comment]
I've sent PRs for the other issues. Especially with the fix for #4 I can use this tool again, using the workaround above. Would still be nice to automate the token creation. Or at least make it more reliable, since it currently looks like the file containing the token is deleted when installing the module from npm again. If nothing else, a note in the README would help.
